### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by DrewBradley**
#### What does this PR do?
Adds comment modal form that appears when the user clicks the add comment button. Adds comment persistence, so the comments are always with their affiliated idea cards.

#### Where should the reviewer start?
Reviewer should focus on lines 29 through 35 of the HTML.  Please review the new comments.js file in its entirety.  In main.js please check lines 61-65, & 77, 128-138, and 181 through the end of the file.
#### How should this be manually tested?
Reviewer should add a few idea cards, then add comments to each idea. Reviewer will then refresh the page to ensure that the comments remain, and the comment count is consistent with the number of comments associated with each idea card.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -136,9 +171,59 @@ function searchIdeas() {
...
+  var text = commentInput.value;
+  var parentId = parseInt(commentInput.id);
+  //save comment to local storage
+  var newComment = new Comment(Date.now(), parentId, text);
+  addCommentToIdea(newComment, parentId)
+  commentInput.value = '';
+  showIdeas(savedIdeas.sort(compareId));
+  newComment.saveToStorage();
+  //loop thru local storage to find corresponding idea
+  //parse idea object
```
Comment (JeffKersting) [inline src/main.js:208]: Remove comments

File: src/main.js [hunk 2]
```diff
@@ -60,6 +74,7 @@ function showIdeas(array) {
         <span class="comment">
           <img src="assets/comment.svg" class="comment-button" alt="comment button">
           <label for="comment-button">Comment</label>
+          <p class="comment-ticker">${array[i].comments.length} ${commentCountText}</p>
```
Comment (colefiscus) [inline src/main.js:77]: I love this feature!

File: src/main.js [hunk 3]
```diff
@@ -7,18 +7,27 @@ var saveButton = document.querySelector(".save-button");
 var searchInput = document.querySelector(".search-input");
 var searchButton = document.querySelector(".search-button");
 var ideaGrid = document.querySelector(".idea-grid");
+var modal = document.querySelector(".modal");
+var commentInput = document.querySelector(".comment-text");
+var commentSubmit = document.querySelector(".submit-comment");
+var commentClose = document.querySelector(".close-button");
```
Comment (JeffKersting) [inline src/main.js:14]: add modal to view comments

**Review by JeffKersting — APPROVED** [review]

File: src/main.js [hunk 4]
```diff
@@ -97,18 +117,25 @@ function favoriteIdea(target) {
...
+        savedIdeas.push(new Idea(idea.title, idea.body, id, idea.star));
+      }else if (!idea.title){
+        savedComments.push(new Comment(idea.id, idea.parentId, idea.text));
+      }
     }
-    showIdeas(savedIdeas);
+    for (var i = 0; i < savedComments.length; i++){
+      addCommentToIdea(savedComments[i], savedComments[i].parentId);
+    };
+    showIdeas(savedIdeas.sort(compareId));
```
Comment (colefiscus) [inline src/main.js:138]: This seems like it works well - I could def use a review/explanation of the process from adding a comment until the end.

File: src/main.js [hunk 5]
```diff
@@ -136,9 +171,59 @@ function searchIdeas() {
...
+
+  //save back to storage
+
+  //on SUBMIT close comment form modal, open comment display modal
+
+  //seperate function to push comment to idea.comment array
+
+  //seperate function to populate the comments of an idea.
+
+  console.log(text);
```
Comment (colefiscus) [inline src/main.js:220]: DELETE!

**Review by colefiscus — APPROVED** [review]
We're almost done!

**Pull request merged**